pyproject.toml

```diff
@@ -61,5 +63,20 @@ build-backend = "maturin"
 module-name = "granian._granian"
 bindings = "pyo3"
 
+[tool.ruff]
+line-length = 120
+extend-select = ['Q', 'RUF100', 'C90', 'I']
+extend-ignore = ['E501']
+flake8-quotes = { inline-quotes = 'single', multiline-quotes = 'double' }
+mccabe = { max-complexity = 13 }
+isort = { combine-as-imports = true, lines-after-imports = 2, known-first-party = ['granian', 'tests'] }
```

Comment:
I would suggest doing it more meticulously than just selecting everything and ignoring a few.

Something like the following:

```suggestion
[tool.ruff]
line-length = 120
extend-select = ['Q', 'RUF100', 'C90', 'I']
extend-select = [
    # E and F are enabled by default
    "B",     # flake8-bugbear
    "C4",    # flake8-comprehensions
    "C90",   # mccabe
    "D",     # pydocstyle
    "RUF100" # ruff (unused noqa)
    "I",     # isort
    "N",     # pep8-naming
    "S",     # flake8-bandit
    "W",     # pycodestyle
]
ignore = [
    "D1",   # public-* (use your best judgement)
    "D203", # one-blank-line-before-class
    "D213", # multi-line-summary-second-line
    "D215", # section-underline-not-over-indented
    "D400", # ends-in-period (in favour of D415)
    "D401", # non-imperative-mood
    "D404", # no-this-prefix
    "D406", # new-line-after-section-name
    "D407", # dashed-underline-after-section
    "E501", # line-too-long (black decides)
]

[tool.ruff.per-file-ignores]
"__init__.py" = ["E402"]
"tests/**" = ["S101", "S105"]

[tool.ruff.mccabe]
max-complexity = 13

[tool.ruff.isort]
known-first-party = ["granian", "tests"]
lines-after-imports = 2
lines-between-types = 1
```

And it can grow from here, by necessity.